### PR TITLE
MeleeLifesteal Upgrade

### PR DESCRIPTION
**Pull request by OtakuFreak13**
Please test if this is an okay upgrade. 
Maybe try it out and see if it makes melee more viable. Also check if you think the pricing and health steal is adequate.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+UpgradeMeleeLifeSteal::~UpgradeMeleeLifeSteal()
+{
+}
+
+void UpgradeMeleeLifeSteal::IncreaseLevel()
+{
+	m_Level++;
+	m_PercentageGain = 0.05f + 0.05f * m_Level; // Starts at 10% and increases by 5% per level
+	m_Price = 300 * pow(m_Level + 1, 2);
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp]: It gets kinda pricy very fast :> tbh I have lost track of how expensive other upgrades get at the moment so maybe it's consistent with other prices.

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+}
+
+void UpgradeMeleeLifeSteal::ApplyBoughtUpgrade()
+{
+}
+
+std::string UpgradeMeleeLifeSteal::GetDescription(unsigned int level)
+{
+	return "Melee Lifesteal: Converts a " + std::to_string(static_cast<int>((0.05f + 0.05f * level) * 100)) + "\% of melee damage done into own health.";
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp]: remove the "a" after Converts.

File: StortSpelProjekt/Game/src/main.cpp [hunk 3]
```diff
@@ -765,7 +765,7 @@ Scene* ShopScene(SceneManager* sm)
     mc->SetDrawFlag(FLAG_DRAW::DRAW_OPAQUE | FLAG_DRAW::GIVE_SHADOW);
     tc = entity->GetComponent<component::TransformComponent>();
     tc->GetTransform()->SetScale(50, 1, 50);
-    tc->GetTransform()->SetPosition(0.0f, 0.0f, 0.0f);
+    tc->GetTransform()->SetPosition(10.0f, 0.0f, 0.0f);
```
Comment (kattalainen) [inline StortSpelProjekt/Game/src/main.cpp]: ? :>

**Review by AntonAsbrink — DISMISSED** [review]

File: StortSpelProjekt/Game/src/main.cpp (hunk 3 above)
Comment (OtakuFreak13) [reply StortSpelProjekt/Game/src/main.cpp]: Woops, should be fixed now though.

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp (hunk 2 above)
Comment (OtakuFreak13) [reply StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp]: Removed

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp (hunk 1 above)
Comment (OtakuFreak13) [reply StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeMeleeLifeSteal.cpp]: Yeah changed it to m_Price += m_Level * 200
That should make it a bit more afforadable at the first 2-3 levels and then scale pretty well I think.

**Review by AntonAsbrink — APPROVED** [review]

**Review by kattalainen — APPROVED** [review]

**Pull request merged**